ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/Delete.hs

```diff
@@ -0,0 +1,134 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.Delete
+  ( DeletingItem(..)
+  , DeletionResult(..)
+  , runDelete
+  ) where
+
+import Universum
+
+import qualified Data.Text as T
+import Formatting
+
+import Graphics.UI.Qtah.Signal (connect_)
+import Graphics.UI.Qtah.Widgets.QSizePolicy (QSizePolicyPolicy(..))
+
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QHBoxLayout as QHBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QCheckBox as QCheckBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.UI.Qt.UI
+
+data DeletingItem = Wallet | Account
+data DeletionResult = DoDelete | Cancel deriving Eq
+
+data Delete =
+  Delete
+    { delete :: QDialog.QDialog
+    , isSure :: QCheckBox.QCheckBox
+    , itemName :: Text
+    , retypeWidget :: QWidget.QWidget
+    , retypeName :: QLineEdit.QLineEdit
+    , deleteButton :: QPushButton.QPushButton
+    }
+
+initDelete :: DeletingItem -> Text -> IO Delete
+initDelete itemType itemName = do
+  delete <- QDialog.new
+  layout <- createLayout delete
+
+  let headerString = toString . T.toUpper $ sformat ("delete " % itemTypeFormat) itemType
+
+  QWidget.setWindowTitle delete headerString
+
+  header <- QLabel.newWithText headerString
+  addHeader layout header
+
+  warningLabel <- QLabel.newWithText . toString $ sformat
+    ("Do you really want to delete <b>" % stext % "</b> " % itemTypeFormat % "?") itemName itemType
+  QBoxLayout.addWidget layout warningLabel
+
+  isSure <- QCheckBox.new
+  QWidget.setSizePolicyRaw isSure Maximum Maximum
+  isSureLabel <- QLabel.newWithText . toString $ sformat
+    ("Make sure you have access to backup before continuing. \
+     \Otherwise you will lose all your funds connected to this " % itemTypeFormat % ".")
+    itemType
+  QLabel.setWordWrap isSureLabel True
+  QWidget.setMinimumSizeRaw isSureLabel 600 30
+  isSureLayout <- QHBoxLayout.new
+  QBoxLayout.addWidget isSureLayout isSure
+  QBoxLayout.addWidget isSureLayout isSureLabel
+  QBoxLayout.addLayout layout isSureLayout
+
+  -- TODO handle empty name case
```

Comment:
See question re `Maybe`. If we were to allow empty names, we'd need to disable retype field entirely in those cases, I guess.

Reply:
`Maybe` is a mistake it seems.

Reply:
Okay, I'll drop it later.